fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -0,0 +1,373 @@
+
+        # run similarity
+        return self.sort_by_similarity(sample["id"], k, dist_field=dist_field)
+
+    def _sample_split(self, sample):
+        sample_split = set(self.config.split_tags) & set(sample.tags)
+        if len(sample_split) > 1:
+            raise ValueError("sample belongs to multiple splits.")
+        if len(sample_split) == 0:
+            raise ValueError(f"sample is not part of any split!")
+        sample_split = sample_split.pop()
+        return sample_split
+
+    def _tags_to_search(self, sample_split):
+        tags_to_search = copy(self.config.split_tags)
+        tags_to_search.remove(sample_split)
+        return tags_to_search
+
+
+###
+
+### HASH BACKEND
+
+_HASH_METHODS = ["filepath", "image"]
+
+
+class LeakySplitsHashConfig(fob.BrainMethodConfig, LeakySplitsConfigInterface):
+    """
+
+    Args:
+        hash_field (None): string, field to write hashes into
+    """
+
+    def __init__(
+        self,
+        split_views=None,
+        split_field=None,
+        split_tags=None,
+        method="filepath",
+        hash_field=None,
+        **kwargs,
+    ):
+        self._method = method
+        self.hash_field = hash_field
+        LeakySplitsConfigInterface.__init__(
+            self,
+            split_views=split_views,
+            split_field=split_field,
+            split_tags=split_tags,
+        )
+        fob.BrainMethodConfig.__init__(self, **kwargs)
+
+    @property
+    def method(self):
+        return self._method
+
+
+class LeakySplitsHash(fob.BrainMethod):
+    def initialize(self, samples, brain_key):
+        return LeakySplitsHashIndex(
+            samples, self.config, brain_key, backend=self
+        )
+
+
+class LeakySplitsHashIndex(fob.BrainResults, LeakySplitIndexInterface):
+    """ """
+
+    def __init__(self, samples, config, brain_key, backend):
+        fob.BrainResults.__init__(
+            self, samples, config, brain_key, backend=backend
+        )
+        LeakySplitIndexInterface.__init__(self)
+        self._hash2ids = defaultdict(list)
+        self.split_views = _to_views(
+            samples,
+            self.config.split_views,
+            self.config.split_field,
+            self.config.split_tags,
+        )
+        self._dataset = samples._dataset
+        self._compute_hashes(samples)
+
+    @property
+    def _hash_function(self):
+        if self.config.method == "filepath":
+            return fou.compute_filehash
+
+        elif self.config.method == "image":
+            return LeakySplitsHashIndex._image_hash
+
+    def _compute_hashes(self, samples):
+        for s in samples.iter_samples():
+            hash = str(self._hash_function(s["filepath"]))
+            self._hash2ids[hash].append(s["id"])
+            if self.config.hash_field:
+                s[self.config.hash_field] = hash
+                s.save()
+
+    @staticmethod
+    def _image_hash(image, hash_size=24):
```

Comment:
I think image hashing is a generally useful enough function that it should be pulled out. For example, the `compute_exact_duplicates` method uses [this](https://github.com/voxel51/fiftyone-brain/blob/2374ef5b20985ad65b38665d55effd1cce725a7a/fiftyone/brain/internal/core/duplicates.py#L93) hash but could also easily be changed to use a image hash.